In [3]:
import pandas as pd

In [4]:
data = pd.read_csv('/Users/danielmorantes/PythonP/Analítica_Comp_Uniandes/Proyecto1/Proyecto1_Analcomp/DataAWS.csv')
data["periodo"][0]
valor = data["periodo"][0]+data["periodo"][1]
valor

np.int64(40294)

In [5]:
#Initial view of the data
raw_data = pd.read_csv('/Users/danielmorantes/PythonP/Analítica_Comp_Uniandes/Proyecto1/Proyecto1_Analcomp/DataAWS.csv')
raw_data.head()

,periodo,estu_tipodocumento,estu_consecutivo,cole_area_ubicacion,cole_bilingue,cole_calendario,cole_caracter,cole_cod_dane_establecimiento,cole_cod_dane_sede,cole_cod_depto_ubicacion,...,fami_tienecomputador,fami_tieneinternet,fami_tienelavadora,desemp_ingles,punt_ingles,punt_matematicas,punt_sociales_ciudadanas,punt_c_naturales,punt_lectura_critica,punt_global
0,20162,TI,SB11201620604627,URBANO,N,A,TÉCNICO/ACADÉMICO,115238000698,115238000698,15,...,NaN,NaN,NaN,A2,61.00,59.00,63.0,57.0,56.0,295.0
1,20132,TI,SB11201320007175,URBANO,N,A,TÉCNICO/ACADÉMICO,315762000076,315762000076,15,...,Si,Si,No,A1,43.00,52.00,NaN,NaN,NaN,NaN
2,20102,CC,SB11201020409042,URBANO,N,A,TÉCNICO,115790000031,115790000031,15,...,Si,Si,Si,A-,38.53,45.38,NaN,NaN,NaN,NaN
3,20142,TI,SB11201420116599,RURAL,N,A,TÉCNICO/ACADÉMICO,215759000524,215759000524,15,...,Si,No,No,A1,52.00,55.00,54.0,60.0,54.0,277.0
4,20152,TI,SB11201520349379,URBANO,N,A,TÉCNICO,115835000283,115835000283,15,...,No,No,Si,A1,55.00,52.00,50.0,53.0,47.0,254.0


In [6]:
raw_data.shape

(215234, 51)

In [7]:
#Colums names
raw_data.columns

Index(['periodo', 'estu_tipodocumento', 'estu_consecutivo',
       'cole_area_ubicacion', 'cole_bilingue', 'cole_calendario',
       'cole_caracter', 'cole_cod_dane_establecimiento', 'cole_cod_dane_sede',
       'cole_cod_depto_ubicacion', 'cole_cod_mcpio_ubicacion',
       'cole_codigo_icfes', 'cole_depto_ubicacion', 'cole_genero',
       'cole_jornada', 'cole_mcpio_ubicacion', 'cole_naturaleza',
       'cole_nombre_establecimiento', 'cole_nombre_sede',
       'cole_sede_principal', 'estu_cod_depto_presentacion',
       'estu_cod_mcpio_presentacion', 'estu_cod_reside_depto',
       'estu_cod_reside_mcpio', 'estu_depto_presentacion', 'estu_depto_reside',
       'estu_estadoinvestigacion', 'estu_estudiante', 'estu_fechanacimiento',
       'estu_genero', 'estu_mcpio_presentacion', 'estu_mcpio_reside',
       'estu_nacionalidad', 'estu_pais_reside', 'estu_privado_libertad',
       'fami_cuartoshogar', 'fami_educacionmadre', 'fami_educacionpadre',
       'fami_estratovivienda', 'fami_perso

In [8]:
#Nan values
raw_data.isna().sum()

periodo                              0
estu_tipodocumento                   0
estu_consecutivo                     0
cole_area_ubicacion                 80
cole_bilingue                    17277
cole_calendario                     49
cole_caracter                      980
cole_cod_dane_establecimiento        0
cole_cod_dane_sede                   0
cole_cod_depto_ubicacion             0
cole_cod_mcpio_ubicacion             0
cole_codigo_icfes                 1616
cole_depto_ubicacion                 0
cole_genero                          0
cole_jornada                         0
cole_mcpio_ubicacion                 0
cole_naturaleza                      0
cole_nombre_establecimiento          0
cole_nombre_sede                     0
cole_sede_principal                  0
estu_cod_depto_presentacion          0
estu_cod_mcpio_presentacion          0
estu_cod_reside_depto             1234
estu_cod_reside_mcpio             1234
estu_depto_presentacion              0
estu_depto_reside        

In [9]:
raw_data.isna().sum(axis=0)/raw_data.shape[0]*100

periodo                           0.000000
estu_tipodocumento                0.000000
estu_consecutivo                  0.000000
cole_area_ubicacion               0.037169
cole_bilingue                     8.027078
cole_calendario                   0.022766
cole_caracter                     0.455318
cole_cod_dane_establecimiento     0.000000
cole_cod_dane_sede                0.000000
cole_cod_depto_ubicacion          0.000000
cole_cod_mcpio_ubicacion          0.000000
cole_codigo_icfes                 0.750811
cole_depto_ubicacion              0.000000
cole_genero                       0.000000
cole_jornada                      0.000000
cole_mcpio_ubicacion              0.000000
cole_naturaleza                   0.000000
cole_nombre_establecimiento       0.000000
cole_nombre_sede                  0.000000
cole_sede_principal               0.000000
estu_cod_depto_presentacion       0.000000
estu_cod_mcpio_presentacion       0.000000
estu_cod_reside_depto             0.573329
estu_cod_re

In [10]:
raw_data["periodo"].unique()

array([20162, 20132, 20102, 20142, 20152, 20172, 20194, 20122, 20112,
       20141, 20224, 20111, 20201, 20101, 20151, 20121, 20131, 20181,
       20211, 20221])

In [12]:
cols = ["punt_lectura_critica","punt_c_naturales","punt_sociales_ciudadanas","punt_ingles","punt_matematicas","punt_global"]
raw_data.groupby("periodo")[cols].count()


,punt_lectura_critica,punt_c_naturales,punt_sociales_ciudadanas,punt_ingles,punt_matematicas,punt_global
periodo,,,,,,
20101,0,0,0,658,658,0
20102,0,0,0,20202,20202,0
20111,0,0,0,487,487,0
20112,0,0,0,20235,20235,0
20121,0,0,0,460,460,0
20122,0,0,0,22352,22352,0
20131,0,0,0,340,340,0
20132,0,0,0,18109,18109,0
20141,0,0,0,307,307,0


no hay registros para puntajes antes de 20141, por lo que se eliminan estos registros, no son relevantes ni aportan valor en los análisis.

In [14]:
raw_data1 = raw_data[raw_data["periodo"] >= 20142]
raw_data1.shape

(132084, 51)

In [15]:
raw_data1.groupby("periodo")[cols].count()

,punt_lectura_critica,punt_c_naturales,punt_sociales_ciudadanas,punt_ingles,punt_matematicas,punt_global
periodo,,,,,,
20142,17325,17325,17325,17325,17325,17325
20151,299,299,299,299,299,299
20152,16538,16538,16538,16538,16538,16538
20162,16840,16840,16840,16840,16840,16840
20172,16872,16872,16872,16872,16872,16872
20181,87,87,87,87,87,87
20194,33474,33474,33474,33474,33474,33474
20201,94,94,94,94,94,94
20211,65,65,65,64,65,65


In [16]:
raw_data1.isna().sum(axis=0)/raw_data.shape[0]*100

periodo                          0.000000
estu_tipodocumento               0.000000
estu_consecutivo                 0.000000
cole_area_ubicacion              0.000000
cole_bilingue                    8.750492
cole_calendario                  0.000000
cole_caracter                    0.602647
cole_cod_dane_establecimiento    0.000000
cole_cod_dane_sede               0.000000
cole_cod_depto_ubicacion         0.000000
cole_cod_mcpio_ubicacion         0.000000
cole_codigo_icfes                0.000000
cole_depto_ubicacion             0.000000
cole_genero                      0.000000
cole_jornada                     0.000000
cole_mcpio_ubicacion             0.000000
cole_naturaleza                  0.000000
cole_nombre_establecimiento      0.000000
cole_nombre_sede                 0.000000
cole_sede_principal              0.000000
estu_cod_depto_presentacion      0.000000
estu_cod_mcpio_presentacion      0.000000
estu_cod_reside_depto            0.046940
estu_cod_reside_mcpio            0

In [17]:
raw_data1.describe()

,periodo,cole_cod_dane_establecimiento,cole_cod_dane_sede,cole_cod_depto_ubicacion,cole_cod_mcpio_ubicacion,cole_codigo_icfes,estu_cod_depto_presentacion,estu_cod_mcpio_presentacion,estu_cod_reside_depto,estu_cod_reside_mcpio,punt_ingles,punt_matematicas,punt_sociales_ciudadanas,punt_c_naturales,punt_lectura_critica,punt_global
count,132084.000000,1.320840e+05,1.320840e+05,132084.0,132084.000000,132084.000000,132084.000000,132084.000000,132022.000000,132022.000000,132081.000000,132084.000000,132084.000000,132084.000000,132084.000000,132084.000000
mean,20181.860415,1.915744e+11,1.937988e+11,15.0,15410.978870,131007.155575,15.738242,16105.002771,15.102581,15506.584986,51.659785,54.356743,51.333053,52.838126,53.735388,264.789104
std,28.855334,9.073463e+10,9.180739e+10,0.0,288.718222,210381.867718,6.614105,6610.937236,2.192938,2210.896201,10.940840,11.219249,11.095774,9.630691,9.751058,46.214481
min,20142.000000,1.150010e+11,1.150010e+11,15.0,15001.000000,4747.000000,5.000000,5001.000000,5.000000,5001.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,20152.000000,1.153320e+11,1.153670e+11,15.0,15176.000000,5595.000000,15.000000,15176.000000,15.000000,15176.000000,44.000000,47.000000,43.000000,46.000000,47.000000,231.000000
50%,20172.000000,1.158040e+11,1.158140e+11,15.0,15407.000000,47316.000000,15.000000,15299.000000,15.000000,15403.000000,51.000000,54.000000,51.500000,53.000000,54.000000,263.000000
75%,20194.000000,3.150010e+11,3.150010e+11,15.0,15740.000000,132019.000000,15.000000,15753.000000,15.000000,15720.000000,58.000000,62.000000,59.000000,60.000000,61.000000,297.000000
max,20224.000000,4.158390e+11,5.154690e+11,15.0,15897.000000,783480.000000,99.000000,99001.000000,85.000000,85225.000000,100.000000,100.000000,100.000000,100.000000,100.000000,471.000000
